### PR TITLE
Add option start_high to Clock.start().

### PR DESCRIPTION
**Pull request by cmarqu**
This comes handy for cases like testpattern generation where you need a bit more control over the clock phase. The default behaviour doesn't change.
Refs #979.

### CODE REVIEW

File: cocotb/clock.py [hunk 1]
```diff
@@ -75,33 +75,43 @@ def __init__(self, signal, period, units=None):
...
+                self.signal <= 1
+                yield t
+                self.signal <= 0
+                yield t
+        else:
+            for _ in it:
+                self.signal <= 0
+                yield t
+                self.signal <= 1
+                yield t
```
Comment (eric-wieser) [inline cocotb/clock.py]: Perhaps clearer, and definitely shorter, as:
```python
for _ in it:
    self.signal <= int(start_high)
    yield t
    self.signal <= int(not start_high)
    yield t
```

Comment (cmarqu) [reply cocotb/clock.py]: I was wary that this might incur an additional runtime overhead, so played it safe.

Comment (eric-wieser) [reply cocotb/clock.py]: Possibly worth a comment like "branch outside for loop for performance"

Comment (cmarqu) [reply cocotb/clock.py]: Good point, I have done so now.

**Comment by themperek** [comment]
Thank you @cmarqu and @eric-wieser

**Pull request merged**